packages/core/data-transfer/src/file/providers/source/index.ts

```diff
@@ -81,18 +83,33 @@ class LocalFileSourceProvider implements ISourceProvider {
       }
       throw new ProviderInitializationError(`File '${filePath}' is not a valid Strapi data file.`);
     }
+
+    if (!this.#metadata) {
+      throw new ProviderInitializationError('Could not load metadata from Strapi data file.');
+    }
   }
 
-  getMetadata() {
-    // TODO: need to read the file & extract the metadata json file
-    // => we might also need to read the schema.jsonl files & implements a custom stream-check
+  async #loadMetadata() {
     const backupStream = this.#getBackupStream();
-    return this.#parseJSONFile<IMetadata>(backupStream, METADATA_FILE_PATH);
+    this.#metadata = await this.#parseJSONFile<IMetadata>(backupStream, METADATA_FILE_PATH);
+    return this.#metadata;
```

Comment:
Does it make sense to return the value from a "loader" function?

Reply:
I don't think it hurts, but I can also remove it to ensure we always use getMetadata when it's needed.